lib/gateway/server.js

```diff
@@ -20,9 +20,16 @@ function createTlsServer (httpsConfig, app) {
   let defaultCert = null;
   let sniCerts = [];
 
-  for (let el in httpsConfig.tls) {
-    let domain = el;
-    let certPaths = httpsConfig.tls[el];
+  let domainCount = httpsConfig.tls.length;
+  let domainNames = [];
+
+  httpsConfig.tls.forEach(domainObj => {
```

Comment:
Can this be collapsed to something like:

```js
const domainNames = httpsConfig.tls.map(domainObj => Object.getOwnPropertyNames(domainObj));
```

@DrMegavolt I think you worked on this in the past.  What behavior should we expect if no `tls` configuration exists?  Should we log a warning?  Exit with a config validation error?  Thoughts?